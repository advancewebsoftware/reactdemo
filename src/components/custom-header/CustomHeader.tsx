import React from 'react';
import {Images} from '../../assets';
import {styles} from './CustomHeaderStyle';
import {FlashList} from '@shopify/flash-list';
import {homeScreenString} from '../../constants';
import {Colors, moderateScale} from '../../theme';
import {Image, Text, TouchableOpacity, View} from 'react-native';
import {CategoriesType} from '../../types';

interface CustomHeadertype {
  isHome: boolean;
  onBackPress?: () => void;
  selectedCategory?: string;
  onSearchPress?: () => void;
  onSelectCategory?: (item: CategoriesType) => void;
}

const CustomHeader = ({
  isHome,
  onBackPress,
  onSearchPress,
  onSelectCategory,
  selectedCategory,
}: CustomHeadertype): React.JSX.Element => {
  const category: CategoriesType[] = [
    'All',
    'health',
    'sports',
    'science',
    'business',
    'technology',
    'entertainment',
  ];

  return (
    <View style={styles.container}>
      {isHome ? (
        <>
          <View style={styles.innerView}>
            <Image
              source={Images.logo_ic}
              style={styles.logoImage}
              resizeMode="contain"
            />
            <Text style={styles.text}>{homeScreenString.todayNews}</Text>
            <TouchableOpacity onPress={onSearchPress}>
              <Image source={Images.search_ic} style={styles.searchImage} />
            </TouchableOpacity>
          </View>
          <FlashList
            horizontal
            data={category}
            showsHorizontalScrollIndicator={false}
            renderItem={({item}: {item: CategoriesType}) => {
              return (
                <TouchableOpacity
                  onPress={() => onSelectCategory && onSelectCategory(item)}
                  style={[
                    styles.horizontalList,
                    {
                      backgroundColor:
                        selectedCategory === item ? Colors.red : Colors.blue9,
                    },
                  ]}>
                  <Text
                    style={{color: Colors.white, fontSize: moderateScale(15)}}>
                    {item}
                  </Text>
                </TouchableOpacity>
              );
            }}
          />
        </>
      ) : (
        <TouchableOpacity onPress={onBackPress}>
          <Image source={Images.arrow_ic} style={styles.arrowImage} />
        </TouchableOpacity>
      )}
    </View>
  );
};

export default CustomHeader;
