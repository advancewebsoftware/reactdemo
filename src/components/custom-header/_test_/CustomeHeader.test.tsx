import React from 'react';
import {render} from '@testing-library/react-native';
import CustomHeader from '../CustomHeader';

const component = <CustomHeader isHome={false} />;

describe('CustomHeader', () => {
  it('renders correctly', () => {
    const {toJSON} = render(component);
    expect(toJSON()).toMatchSnapshot();
  });
});
