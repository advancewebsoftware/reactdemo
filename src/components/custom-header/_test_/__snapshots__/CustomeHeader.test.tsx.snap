// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomHeader renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#141831",
      "padding": 30,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/images/arrow.png",
        }
      }
      style={
        {
          "height": 39.42857142857143,
          "tintColor": "#FFFFFF",
          "width": 48,
        }
      }
    />
  </View>
</View>
`;
